src/components/ChatComponents/ChatIcons.tsx

```diff
@@ -62,25 +64,26 @@ const ChatIcons: React.FC<ChatIconsProps> = ({
     const selectedModel = event.target.value;
     setCurrentModel(event.target.value);
 
-    if (selectedModel === ChatModelDisplayNames.CLAUDE) {
-      // Start the proxy server when CLAUDE is selected
-      await proxyServer.startProxyServer('https://api.anthropic.com/');
+    // Start proxy server based on the selected model & settings
+    const proxyServerURL = proxyServer.getProxyURL(selectedModel);
+    if (proxyServerURL) {
+      await proxyServer.startProxyServer(proxyServerURL, selectedModel !== ChatModelDisplayNames.CLAUDE);
```

Comment:
The `selectedModel !== ChatModelDisplayNames.CLAUDE` check here and on line 79 may not be needed. 

If proxy support works for Anthropic when rewriting paths, then the 2nd argument can be removed from `startProxyServer`